tests/test_dispatch.py

```diff
@@ -37,3 +44,9 @@ def test_dispatch():
 
     endpoint = dispatcher.handlers.get("/hello")[0]
     assert endpoint == dummy_endpoint
+
+
+def test_dispatch_unexpected_message():
+    dispatcher = RequestDispatcher()
+    with pytest.raises(FatalProtocolError):
+        dispatcher.handle(CallResponseMessage(), None)
```

Comment:
Maybe assert that the error message includes what we expect? General convention is:

    with pytest.raises(..) as exc_info:
        ...

    assert 'Received a bad call req cont' in str(exc_info)